scopes/component/component/ui/use-component.tsx

```diff
@@ -13,18 +12,13 @@ export type Component = {
   componentDescriptor?: ComponentDescriptor;
 };
 
-type ComponentRoute = {
-  componentId?: string;
-};
-
-export function useComponent(host: string, id?: ComponentID): Component {
-  const {
-    params: { componentId },
-  } = useRouteMatch<ComponentRoute>();
+export function useComponent(host: string, id?: string | ComponentID): Component {
```

Comment:
to fix
- [ ]should only get a componentID, and use its version if available, else use from params